### ISSUE TITLE
Leaves decaying can break Auction signs.

### ISSUE THREAD

**Issue by NullCase** [RoboMWM/GPAuctions#68]
An interesting bug.

Basically, if an AutoAuction sign appears on a leaf block and the leaves are caused to decay (by cutting down the tree) then the Auction sign breaks, which prevents further bidding.

I don't know what to do about this right now. Just reporting and going to think on it.

**Comment by NullCase** [comment]
If the server isn't restarted, it seems like the auction would close as intended... but I don't know and plan to test it out before our next update.

**Comment by RoboMWM** [comment]
Hmm, I could implement something to bypass leaves when placing the sign if desired

**Comment by NullCase** [comment]
a bypass. So it sounds like that would trigger when creating an AutoAuction.

It seems like that would work if we go more general. Because user created auctions would still offer an exploit if the auction sign were placed on leaves. It's an edge case, but maybe the general solution is no more complicated. So lets go a step further.

What's the more general solution? stop auction signs being placed on leaves? Perhaps ice blocks qualify...

Or is it better to just prevent auction signs from breaking by environmental effects? Is that an option?

**Comment by NullCase** [comment]
btw I have dev money available for this fix.

Also, if we do this one I want to add a quality of life update: The world name included when auctions are broadcast. I'll draw up an issue. It'll match GPR's format.

**Comment by RoboMWM** [comment]
well it doesn't care where the sign is, as long as it's in the claim. 

When creating an auction automatically it has to place the sign (since there's no player there to do it), and right now it just does the easiest thing - place the sign on the highest non-air block in the bottom-left corner of the claim. The "highest non-air block" check could also include leaves, such that it'll be "highest non-air, non-leaf" block. Or - it could add and/or replace the block that it stands on, therefore ensuring that it isn't on an "unstable" block.

**Comment by NullCase** [comment]
It seems like there's gotta be something elegant which solves this without creating more problems.

For example, with a non-air, non-leaf policy, autoauction signs might appear inside the foliage of a tree on a log, out of reach. 

For example, if placed on an ice block, the ice can melt breaking a sign. And it seems like Mojang will continue to add blocks with complex behavior like this.

**Comment by NullCase** [comment]
Can I make autoauction signs ignore environmental effects?

So if the leaves decay or something it just floats there instead of breaking?

**Comment by RoboMWM** [comment]
> Can I make autoauction signs ignore environmental effects?

No, unless we store the sign location, monitoring for and canceling any BlockChangeEvents, which isn't really ideal.

> Or - it could add and/or replace the block that it stands on, therefore ensuring that it isn't on an "unstable" block.

Sounds like it would resolve

> if placed on an ice block, the ice can melt breaking a sign. And it seems like Mojang will continue to add blocks with complex behavior like this.

And if it's necessary, we could add something to clear out a specified/configurable set of blocks around the sign if this is ever becomes a problem:

> autoauction signs might appear inside the foliage of a tree on a log, out of reach.

**Comment by NullCase** [comment]
I'm taking a break for a few days. I'll be able to respond after that, next weekend seems like.

Thanks for the quick responses.
